Client.cpp

```diff
@@ -0,0 +1,169 @@
+#include "Client.hpp"
+
+Client::Client()
+{
+	// Initialize Winsock
+	if (WSAStartup(MAKEWORD(2, 2), &_ws) == SOCKET_ERROR)
+	{
+		std::cout << "WSA Failed to Initialize\n";
+		exit(EXIT_FAILURE);
+	}
+	else
+	{
+		std::cout << "WSA opened Succesfully\n";
+	}
+	//Initialize the socket
+	_client_fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	if (_client_fd == SOCKET_ERROR)
+	{
+		std::cout << "Failed to Initialize the socket\n";
+		exit(EXIT_FAILURE);
+	}
+	else
+	{
+		std::cout << "The socket opened Succesfully\n";
+		std::cout << "============================\n";
+	}
+
+	// Set up the sockaddr_in structure
+	_clientAddr.sin_family = AF_INET;
+	_clientAddr.sin_port = htons(PORT);
+	inet_pton(AF_INET,IP,&_clientAddr.sin_addr); //convert IPv4 addresses from text to binary 
+
+	int connectResult = connect(_client_fd, (sockaddr*)&_clientAddr, sizeof(_clientAddr));
```

Comment:
You use c casting here read about `reinterpret_cast`, `static_cast`, and `dynamic_cast`  in the std lib, and tell me which one you need to use here